backend_new/src/services/multiselect-question.service.ts

```diff
@@ -0,0 +1,195 @@
+import { MultiselectQuestionUpdate } from '../dtos/multiselect-questions/multiselect-question-update.dto';
+import { MultiselectQuestionCreate } from '../dtos/multiselect-questions/multiselect-question-create.dto';
+import { mapTo } from '../utilities/mapTo';
+import { SuccessDTO } from '../dtos/shared/success.dto';
+import { Prisma } from '@prisma/client';
+import { buildFilter } from '../utilities/build-filter';
+import { MultiselectQuestionFilterKeys } from '../enums/multiselect-questions/filter-key-enum';
+import { MultiselectQuestionQueryParams } from '../dtos/multiselect-questions/multiselect-question-query-params.dto';
+
+const view: Prisma.MultiselectQuestionsInclude = {
+  jurisdictions: true,
+};
+
+/*
+  this is the service for multiselect questions
+  it handles all the backend's business logic for reading/writing/deleting multiselect questione data
+*/
+@Injectable()
+export class MultiselectQuestionService {
+  constructor(private prisma: PrismaService) {}
+
+  /*
+    this will get a set of multiselect questions given the params passed in
+  */
+  async list(
+    params: MultiselectQuestionQueryParams,
+  ): Promise<MultiselectQuestion[]> {
+    const rawMultiselectQuestions =
+      await this.prisma.multiselectQuestions.findMany({
+        include: view,
+        where: this.buildWhere(params),
+      });
+    return mapTo(MultiselectQuestion, rawMultiselectQuestions);
+  }
+
+  /*
+    this will build the where clause for list()
+  */
+  buildWhere(
+    params: MultiselectQuestionQueryParams,
+  ): Prisma.MultiselectQuestionsWhereInput {
+    const filters: Prisma.MultiselectQuestionsWhereInput[] = [];
+    if (params?.filter?.length) {
+      params.filter.forEach((filter) => {
+        if (MultiselectQuestionFilterKeys.jurisdiction in filter) {
+          const builtFilter = buildFilter({
+            $comparison: filter.$comparison,
+            $include_nulls: false,
+            value: filter[MultiselectQuestionFilterKeys.jurisdiction],
+            key: MultiselectQuestionFilterKeys.jurisdiction,
+            caseSensitive: true,
+          });
+          filters.push({
+            OR: builtFilter.map((filt) => ({
+              jurisdictions: {
+                some: {
+                  id: filt,
+                },
+              },
+            })),
+          });
+        } else if (MultiselectQuestionFilterKeys.applicationSection in filter) {
+          const builtFilter = buildFilter({
+            $comparison: filter.$comparison,
+            $include_nulls: false,
+            value: filter[MultiselectQuestionFilterKeys.applicationSection],
+            key: MultiselectQuestionFilterKeys.applicationSection,
+            caseSensitive: true,
+          });
+          filters.push({
+            OR: builtFilter.map((filt) => ({
+              applicationSection: filt,
+            })),
+          });
+        }
+      });
+    }
```

Comment:
Is there ever a case where we would want to get all multiselect questions of applicationSection X and jurisdiction Y? With this setup we can only do one or the other even though the filters imply that we can do both

Reply:
Hey @ludtkemorgan so I think we can still do that, since `MultiselectQuestionQueryParams`'s filter key is an array
so if we wanted to do that we could do [{ jurisdiction: y}, { applicationSection: x}] 

as the params passed in. hopefully that makes some sense

Reply:
Ope, I didn't notice this was all inside of a forEach loop. You are correct. Ignore my comment